public/app/services/settingsService.js

```diff
@@ -1,53 +1,259 @@
 angular.module('kibibitCodeEditor')
 
-.service('SettingsService', ['Fullscreen', function(Fullscreen) {
-  var vm = this;
-
-  /* init */
-  vm.settings = new Settings();
-
-  function Settings() {
-    var settings = this;
-    /* INIT */
-    var cursor = {row: '0', column: '0'};
-    var isFullscreen = false;
-    var currentUndoManager = undefined;
-    var currentEditor = undefined;
-
-    /* EXPOSE SIMPLE VARS */
-    // This are probably better off in state instead of settings. but they're here for now :-)
-    settings.cursor = cursor;
-    settings.currentUndoManager = currentUndoManager;
-    settings.currentEditor = currentEditor;
-
-    settings.__defineSetter__('isFullscreen', function(newValue) {
-
-      console.assert(isBoolean(newValue), {
-        'message': 'isFullscreen should only be a boolean, but was given some other type',
-        'currentValue': isFullscreen,
-        'newValue': newValue
-      });
+.service('SettingsService', [
+  'Fullscreen',
+  'CODE_EDITOR',
+  'TYPE_ERROR_MSGS',
+  function(
+    Fullscreen,
+    CODE_EDITOR,
+    TYPE_ERROR_MSGS) {
+
+    var vm = this;
+
+    /* init */
+    vm.settings = new Settings();
+
+    function Settings() {
+      var settings = this;
+      /* INIT */
+      var cursor = {row: '0', column: '0'};
+      var isFullscreen = false;
+      var currentUndoManager = undefined;
+      var currentEditor = undefined;
+      var editorSettings = new EditorSettings();
+
+      /* EXPOSE SIMPLE VARS */
+      // This are probably better off in state instead of settings. but they're here for now :-)
+      settings.cursor = cursor;
+      settings.currentUndoManager = currentUndoManager;
+      settings.currentEditor = currentEditor;
+      settings.modelist = CODE_EDITOR.MODE_LIST;
+      settings.themelist = CODE_EDITOR.THEME_LIST;
+      settings.editorSettings = editorSettings;
 
-      if (newValue !== currentFullscreenState()) {
-        if (newValue) {
-          Fullscreen.all();
-        } else {
-          Fullscreen.cancel();
+      settings.__defineSetter__('isFullscreen', function(newValue) {
+
+        console.assert(isBoolean(newValue), {
+          'message': TYPE_ERROR_MSGS('isFullscreen', 'boolean',
+            typeof newValue),
+          'currentValue': isFullscreen,
+          'newValue': newValue
+        });
+
+        if (newValue !== currentFullscreenState()) {
+          if (newValue) {
+            Fullscreen.all();
+          } else {
+            Fullscreen.cancel();
+          }
         }
-      }
-    });
+      });
+
+      settings.__defineGetter__('isFullscreen', function() {
+        return currentFullscreenState();
+      });
+
+      function EditorSettings() {
+        var theme = 'monokai';
+        var ruler = 80;
+        var tabWidth = 4;
+        var fontSize = 12;
+        var isGutter = true;
+        var lineWrap = false;
+        var isReadOnly = false;
+        var isSoftTabs = false;
+        var syntaxMode = 'text';
```

Comment:
Don't use this directly. I think it's confusing since you're not sure if `this` is the settingsService, settings object, or the editorSettings object.

please use the same syntax I used for the settings object:
```javascript
function EditorSettings() {
  var editorSettings = this;
  /* INIT */
  var theme = 'monokai';
  var ruler = 80;
  var tabWidth = 4;
  var fontSize = 12;
  var isGutter = true;
  var lineWrap = false;

...
```